### PR TITLE
GLT-2702: mark required fields on all forms

### PR DESCRIPTION
**Pull request by djcooke**

### CODE REVIEW

File: src/main/java/ca/on/oicr/gsi/ramen/data/Product.java [hunk 1]
```diff
@@ -54,52 +56,53 @@ public void setId(long productId) {
...
   }
   
   public String getCatalogNumber() {
     return catalogNumber;
   }
   
   public void setCatalogNumber(String catalogNumber) {
-    this.catalogNumber = catalogNumber.trim();
+    this.catalogNumber = nullOrTrimmed(catalogNumber)
+        ;
```
Comment (callunity) [inline src/main/java/ca/on/oicr/gsi/ramen/data/Product.java]: weird formatting

File: src/main/webapp/pages/stock.jsp [hunk 2]
```diff
@@ -11,11 +11,11 @@
...
-          <td>Order Number</td>
+          <td>Order Number*</td>
           <td><input type="text" id="orderNumber" name="orderNumber"> </td>
         </tr>
         <tr>
-          <td>Date Received</td>
+          <td>Date Received*</td>
           <td><input type="date" id="receiveDate" value="${today}" name="receiveDate"/> </td>
         </tr>
         <tr>
```
Comment (callunity) [inline src/main/webapp/pages/stock.jsp]: Packing Slip should also be marked as required.

Comment (djcooke) [reply src/main/webapp/pages/stock.jsp]: Packing slip isn't required

File: src/main/java/ca/on/oicr/gsi/ramen/data/Product.java (hunk 1 above)
Comment (djcooke) [reply src/main/java/ca/on/oicr/gsi/ramen/data/Product.java]: fixed

**Pull request merged**